tests/Detector.Tests/Go/GoDetectorTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Detector.Go;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+namespace Microsoft.Oryx.Detector.Tests.Go
+{
+    public class GoDetectorTest : IClassFixture<TestTempDirTestFixture>
+    {
+
+        private const string GoFileWithVersion = @"
+            module hello
+
+            go 1.16
+        ";
+
+        private const string GoFileWithNoVersions = @"
+            module hello
+        ";
```

Comment:
I'm not sure how tests are structure in Oryx, but it might be worth having these as individual files in the `Resources` file that are referenced via the `Properties/Resources.resx` file, that way you can easily add new test files without needing to bloat the actual test class with the string implementations.

Reply:
Right, I will look into that. Will also consider parametrization